build_deps.sh

```diff
@@ -14,10 +14,21 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-if [ -d "third_party" ]; then
-  echo "Directory third_party exists, remove this directory before running build_deps.sh."
-  exit 1;
-fi
+build_llvm_with_instcombine_switch=false
+while [ $# -gt 0 ]
+do
+    case "$1" in
+        --enable-instcombine-switch)  build_llvm_with_instcombine_switch=true;;
+        *)  echo "Usage: ./build_deps.sh [--enable-instcombine-switch]" && exit 0;;
+    esac
+    shift
+done
+
+#if [ -d "third_party" ]; then
+#  echo "Directory third_party exists, remove this directory before running build_deps.sh."
+#  exit 1;
+#fi
```

Comment:
you left this check of the third_party directory commented out!

but really, I think it adds complexity to make this patch apply conditionally, I would recommend that we apply it unconditionally